crates/pallet-domains/src/tests.rs

```diff
@@ -133,11 +132,10 @@ impl pallet_domains::HoldIdentifier<Test> for HoldIdentifier {
 }
 
 impl VariantCount for HoldIdentifier {
-    const VARIANT_COUNT: u32 = mem::variant_count::<Self>() as u32;
+    const VARIANT_COUNT: u32 = 10;
```

Comment:
This is wrong though, check the description of the trait. It is supposed to return the same number as number of enum variants. Substrate can't use `mem::variant_count` because they rely on stable compiler, but we can because we use nightly.

Reply:
Yeah, this violates the usage of the trait, but the max number of hold limits is hardcoded to `HoldIdentifier::VARIANT_COUNT` in [pallet-balances](https://github.com/paritytech/polkadot-sdk/blob/216509dbaa2c2941ee75fbcc9a086deab5e2c8a6/substrate/frame/balances/src/lib.rs#L476-L485), which doesn't fit into our usage (even in test) so we need this hack.

Reply:
For the reviewer, see more discussion at https://github.com/subspace/subspace/issues/2674. PR is forced push to add TODO comment.